routing/missioncontrol_state.go

```diff
@@ -209,3 +209,46 @@ func (m *missionControlState) getSnapshot() *MissionControlSnapshot {
 
 	return &snapshot
 }
+
+// importSnapshot takes an existing snapshot and merges it with our current
+// state if the result provided are fresher than our current results.
+func (m *missionControlState) importSnapshot(snapshot *MissionControlSnapshot) {
+	for _, pair := range snapshot.Pairs {
+		fromNode := pair.Pair.From
+		toNode := pair.Pair.To
+
+		results, found := m.getLastPairResult(fromNode)
+		if !found {
+			results = make(map[route.Vertex]TimedPairResult)
+		}
+
+		lastResult := results[toNode]
+
+		failResult := failPairResult(pair.FailAmt)
+		m.importResult(
+			lastResult.FailTime, pair.FailTime, failResult,
+			fromNode, toNode,
+		)
+
+		successResult := successPairResult(pair.SuccessAmt)
+		m.importResult(
+			lastResult.SuccessTime, pair.SuccessTime, successResult,
+			fromNode, toNode,
+		)
+	}
+}
+
+func (m *missionControlState) importResult(currentTs, importedTs time.Time,
+	importedResult pairResult, fromNode, toNode route.Vertex) {
+
+	if currentTs.After(importedTs) {
```

Comment:
Why wouldn't we want to simply overwrite? I see it perfectly valid to just overwrite with an old dump. WDYT?

Reply:
Not sure why would we want to overwrite our existing fresh results with older results? Since the goal in doing this is to get a hotter cache